geode-assembly/src/acceptanceTest/java/org/apache/geode/cache/wan/SeveralGatewayReceiversWithSamePortAndHostnameForSendersTest.java

```diff
@@ -258,7 +258,8 @@ public void testPingsToReceiversWithSamePortAndHostnameForSendersUseOnlyOneMoreC
 
     await().untilAsserted(() -> assertThat(getSenderPoolDisconnects(vm1, senderId)).isEqualTo(0));
 
-    await().untilAsserted(() -> assertThat(getSenderPoolConnects(vm1, senderId)).isEqualTo(4));
+    await().untilAsserted(
+        () -> assertThat(getSenderPoolConnects(vm1, senderId)).isLessThanOrEqualTo(4));
```

Comment:
Why is the number of pool connects not deterministic in this test? Is the fact that it can vary a bug, or expected behaviour?

Reply:
Because it depends on the moment we are checking number of connections, and it can be 3 or 4.

Reply:
But the assertion is in an `await()` so it gets checked repeatedly (every 50 ms by default) until the await times out (which takes 5 minutes). So it seems like the time we're checking the stat can't be the reason.

Reply:
If it can be 3 or 4 then let's do:
`assertThat(getSenderPoolConnects(vm1, senderId)).isIn(3,4)`